tests/integration/tests.mk

```diff
@@ -45,9 +45,14 @@ ifeq ($(_INTEGRATION_TEST_KUBECONFIG),)
     _INTEGRATION_TEST_KUBECONFIG = $(KUBECONFIG)
 endif
 
-# If neither $(INTEGRATION_TEST_KUBECONFIG) nor $(KUBECONFIG) specified, use default.
+# If neither $(INTEGRATION_TEST_KUBECONFIG) nor $(KUBECONFIG) specified, use user default.
 ifeq ($(_INTEGRATION_TEST_KUBECONFIG),)
-    _INTEGRATION_TEST_KUBECONFIG = ~/.kube/config
+    _INTEGRATION_TEST_KUBECONFIG = ${HOME}/.kube/config
+endif
+
+# If running within the build container, use default
+ifeq ($(IN_BUILD_CONTAINER),1)
+    _INTEGRATION_TEST_KUBECONFIG = /home/.kube/config
```

Comment:
isn't HOME always /home here? So do we need this given we already have L50?

Reply:
I suspect the miscommunication is your `$HOME` = `/home`. My $HOME = `/home/sdake`. WIth this second (and more typical) case, L50 is insufficient.

I'd like to rewrite this entire structured conditional from lines 40` - `56`, although I wanted to be fairly surgical so a backport would be feasible.

Reply:
Oh we pass the user HOME? I thought we always set to to /home was the source of confusion

Reply:
For further clarity, /home/.kube/config is where all the config is loaded in the BUILD_WITH_CONTAINER=1 condition.

Reference this flow control:
1. `make test` is run on the host.
1. The host [bind mounts ${HOME}/.kube to the container location /config/.kube]((https://github.com/istio/common-files/blob/master/files/common/scripts/setup_env.sh#L94-L103) and runs the container.
1. The container [entrypoint is executed and copies /config/.kube to /config-copy/.kube](https://github.com/istio/tools/blob/master/docker/build-tools/docker-entrypoint.sh#L24)
1. The container [ copies /config-copy/ to /home within the container](https://github.com/istio/tools/blob/master/docker/build-tools/docker-entrypoint.sh#L55)
1. finally the test targets are run and reference $HOME - which in this case is always /home.

Reply:
We never pass the user's HOME into the container. $HOME in the container environment is hardcoded. However, we do use it outside of the container - which is what this PR and https://github.com/istio/common-files/pull/237 rectify.

Reply:
I suppose we use $HOME inside the container - although $HOME inside the container and outside the container can be different. I am not sure how we can immediately rectify this variable naming problem. Loads of stuff within the container requires a $HOME, but it may be different than the host $HOME.

Reply:
I'm confused by these comments as we don't pass HOME to the container, and we set it to /home in the container. Running locally HOME will be the same as ~.

Reply:
The only thing this changes is if I set _INTEGRATION_TEST_KUBECONFIG as the build container will override it to the user's kubeconfig instead which seems wrong.

Reply:
@ericvn I see what you are saying. The INTEGRATION_TEST_KUBECONFIG is not copied into the container. This is indeed a problem. My general question though is why does this variable exist? We can certainly add logic to setup_env.sh to add a conditional host mount if this variable is set although I still don’t understand the value of this added complexity.